src/use-swr-infinite.ts

```diff
@@ -185,23 +178,37 @@ function useSWRInfinite<Data = any, Error = any>(
   )
 
   // extend the SWR API
-  const size = pageCountRef.current
   const setSize = useCallback(
     arg => {
       if (typeof arg === 'function') {
-        pageCountRef.current = arg(pageCountRef.current)
+        cache.set(pageCountCacheKey, arg(resolvePageCount()))
+        lastPageCountRef.current = arg(resolvePageCount())
       } else if (typeof arg === 'number') {
-        pageCountRef.current = arg
+        cache.set(pageCountCacheKey, arg)
+        lastPageCountRef.current = arg
       }
-      cache.set(pageCountCacheKey, pageCountRef.current)
-      rerender(v => !v)
```

Comment:
I didn't catch why the `rerender` was required, so please let me know if SWR still need to have it.

Reply:
That's because we updated the size by changing a ref: `pageCountRef.current = arg`, which doesn't trigger a re-render (although it could be triggered by `mutate`, that might take a while). But the user needs to get the feedback, for example:

```js
const { size, setSize } = useSWRInfinite(...)

console.log(size)

return <button onClick={() => setSize(size + 1)}>inc</button>
```

When you click the button, it might not output the new page size.

Reply:
Thank you!
In this case, `mutate(v => v)` seems to be enough to trigger a re-render.
https://codesandbox.io/s/swr-infinite-mv98u?file=/src/App.js
The CodeSandbox updates the `size` immediately.